### DISCUSSION TITLE
unknown user error with netatalk2 and Mac OS 8

### DISCUSSION THREAD

**Discussion by 68kmentat** [retronas/discussions#64] [Q&A] [answered]
Has anyone successfully logged in to netatalk2 from Mac OS 8.0?

I am connected to my network via ethernet, my Mac does not pick up the retroafp server automatically but it does recognize it when I manually enter the IP address. I am able to access the server directory as a guest, but when I attempt to log in with my retronas username (which is "retronas") and password (which I changed to be 8 characters long, since that is the maximum that my Mac will let me enter) I get the following error:
`unknown user, log on is disabled or incorrect password. Please retype the name/password or contact the server's administrator.`

I am unfamiliar with AppleTalk and Macs in general, so any help is appreciated

**Comment by danmons** [comment]
The "unknown user" Netatalk2 error is typically because you need to re-run the "username" and "password"  tools in the RetroNAS Global Configuration section again AFTER installing Netatalk2.

Some services in RetroNAS can use the Linux username/password.  For example, Netatalk3 and advanced modern TCP/IP based AppleTalk security protocols can.  However older DDP based AppleTalk protocols require a special hashed password to be written to disk that older computers access directly, which is why the need to re-do this step after installing Netatalk2.

Samba is the same - all SMB protocols need their passwords hashed and stored separately to the Linux system, so users installing Samba must also re-do this process after installing their protocol of choice.  Kind of annoying, but that's how these old protocols work.

That password reset tool in RetroNAS goes looking for all the services it knows about, and sets them (and the Linux user) all with the same username/password across the board.  It needs the specific password hashing tools installed from each project, which is why it has to be re-run after installing each family of protocol that supplies the matching password hashing tools.

**Reply by 68kmentat** [reply]
hmm, I hadn't tried running the username tool -- and I had to change it to something else then back to retronas -- but I had run the password tool already. I've also now run Permissions on .AppleDB and .AppleDesktop to change ownership from root to retronas. Still the same error message.

I finally solved this by manually running sudo /opt/retronas/bin/netatalk2x/bin/afpexpect.sh
I'm guessing that the retronas control panel wasn't running it with sudo and was getting a silent permission denied

**Reply by 68kmentat** [reply]
Looking at `/opt/retronas/bin/netatalk2x/etc/netatalk/afppasswd` also confirms that it was never being written -- it was a 0 byte file until I manually ran the password script with sudo. am I supposed to be running `sudo retronas` instead of just `retronas`? Or is this an issue with the retronas script?

**Reply by 68kmentat** [reply]
Anyway, thank you for your help @danmons. It pointed me in the right direction

**Reply by danmons** [reply]
RetroNAS should handle things like root/sudo elevation for password tools.   If the afppasswd tool isn't writing out, that's a bug at our end.

Thanks for updating this, I'll go bug hunting.

**Reply by sairuk** [reply]
yeh my bug sorry about that, wasn't being run as sudo

https://github.com/danmons/retronas/blob/d5413efcf38d305daf22bb9db1b9fd9366fca92e/scripts/static/update-passwd.sh#L31


have pushed a fix

https://github.com/danmons/retronas/blob/a76d703f8a679ed4d3e048768136bb1baed84b6d/scripts/static/update-passwd.sh#L31

**Comment by 68kmentat** [comment, marked as answer]
solution: afppasswd was not being written and had to be manually set using `sudo /opt/retronas/bin/netatalk2x/bin/afpexpect.sh [username] [password]`
